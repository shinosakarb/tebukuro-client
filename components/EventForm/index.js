// @flow
import React, { Component } from 'react'

type Props = { onSubmit: Function }

export default class EventForm extends Component<Props> {
  constructor(props: Props) {
    super(props)
    this.onSubmitHandler = this.onSubmitHandler.bind(this)
  }

  onSubmitHandler(e: SyntheticEvent<>) {
    e.preventDefault()
    if (this.name && this.description) {
      const params = {
        name: this.name.value,
        description: this.description.value,
      }
      this.props.onSubmit(params)
    }
  }

  onSubmitHandler: Function
  name: ?HTMLInputElement
  description: ?HTMLTextAreaElement

  render() {
    return (
      <div>
        <h3>Event registration form</h3>
        <form onSubmit={this.onSubmitHandler}>
          <div>
            <label htmlFor="name">
              name
              <input type="text" id="name" ref={(input) => { this.name = input }} />
            </label>
          </div>
          <div>
            <label htmlFor="description">
              description
              <textarea id="description" ref={(input) => { this.description = input }} />
            </label>
          </div>
          <div>
            <input type="submit" />
          </div>
        </form>
      </div>
    )
  }
}
