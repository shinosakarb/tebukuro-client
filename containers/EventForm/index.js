// @flow
import React from 'react'
import { connect } from 'react-redux'

export const EventForm = () => (
  <div>This is the EventForm Container.</div>
)

export default connect(null)(EventForm)
