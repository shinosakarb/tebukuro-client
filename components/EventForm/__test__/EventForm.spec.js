import React from 'react'
import { mount, shallow } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import EventForm from '../index'

const onSubmit = jest.fn()

const inputValues = {
  name: 'event1',
  description: 'This is the first event.',
}

describe('EventForm', () => {
  it('render event form component', () => {
    const wrapper = shallow(<EventForm onSubmit={onSubmit} />)
    const tree = shallowToJson(wrapper)

    expect(tree).toMatchSnapshot()
  })

  describe('when submit', () => {
    it('should call onSubmit function with correct argument.', () => {
      const wrapper = mount(<EventForm onSubmit={onSubmit} />)

      const nameInput = wrapper.find('[type="text"]')
      const descriptionInput = wrapper.find('textarea')

      nameInput.instance().value = inputValues.name
      descriptionInput.instance().value = inputValues.description

      wrapper.find('[type="submit"]').simulate('submit')

      expect(onSubmit).toHaveBeenCalledTimes(1)
      expect(onSubmit).toBeCalledWith(inputValues)
    })
  })
})
