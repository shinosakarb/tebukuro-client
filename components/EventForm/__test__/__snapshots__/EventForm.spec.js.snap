// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm render event form component 1`] = `
<div>
  <h3>
    Event registration form
  </h3>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="name"
      >
        name
        <input
          id="name"
          type="text"
        />
      </label>
    </div>
    <div>
      <label
        htmlFor="description"
      >
        description
        <textarea
          id="description"
        />
      </label>
    </div>
    <div>
      <input
        type="submit"
      />
    </div>
  </form>
</div>
`;
